app/gui2/src/components/DocumentationPanel.vue

```diff
@@ -0,0 +1,277 @@
+})
+
+const sections = computed<Sections>(() => {
+  const docs: Docs = documentation.value
+  const fallback = { tags: [], synopsis: [], examples: [] }
+  if ('Function' in docs) {
+    return docs.Function.sections
+  } else if ('Type' in docs) {
+    return docs.Type.sections
+  } else if ('Module' in docs) {
+    return docs.Module.sections
+  } else if ('Local' in docs) {
+    return docs.Local.sections
+  } else {
+    return fallback
+  }
+})
+const methods = computed<FunctionDocs[]>(() => {
+  const docs = documentation.value
+  if ('Module' in docs) {
+    return docs.Module.methods
+  } else if ('Type' in docs) {
+    return docs.Type.methods
+  } else {
+    return []
+  }
+})
+const constructors = computed<FunctionDocs[]>(() => {
+  const docs = documentation.value
+  if ('Type' in docs) {
+    return docs.Type.constructors
+  } else {
+    return []
+  }
+})
+const types = computed<TypeDocs[]>(() => {
+  const docs = documentation.value
+  if ('Module' in docs) {
+    return docs.Module.types
+  } else {
+    return []
+  }
+})
+</script>
+
+<template>
+  <div class="DocumentationPanel">
+    <h1 v-if="'Placeholder' in documentation">{{ documentation.Placeholder }}</h1>
+    <DocsTags v-if="sections.tags.length > 0" :tags="sections.tags" />
+    <DocsSynopsis :sections="sections.synopsis" />
+    <DocsHeader v-if="types.length > 0" kind="types" label="Types" />
+    <DocsList :items="{ Types: types }" @linkClicked="emit('update:selectedEntry', $event)" />
+    <DocsHeader v-if="constructors.length > 0" kind="methods" label="Constructors" />
+    <DocsList
+      :items="{ Constructors: constructors }"
+      @linkClicked="emit('update:selectedEntry', $event)"
+    />
+    <DocsHeader v-if="methods.length > 0" kind="methods" label="Methods" />
+    <DocsList :items="{ Methods: methods }" @linkClicked="emit('update:selectedEntry', $event)" />
+    <DocsHeader v-if="sections.examples.length > 0" kind="examples" label="Examples" />
+    <DocsExamples :examples="sections.examples" />
+  </div>
+</template>
+
+<style scoped>
+.DocumentationPanel {
+  --enso-docs-type-name-color: #9640da;
+  --enso-docs-methods-header-color: #1f71d3;
+  --enso-docs-method-name-color: #1f71d3;
+  --enso-docs-types-header-color: #1f71d3;
+  --enso-docs-examples-header-color: #6da85e;
+  --enso-docs-important-background-color: #edefe7;
+  --enso-docs-info-background-color: #e6f1f8;
+  --enso-docs-example-background-color: #e6f1f8;
+  --enso-docs-background-color: #eaeaea;
+  --enso-docs-text-color: rbga(0, 0, 0, 0.6);
+  --enso-docs-tag-background-color: #dcd8d8;
+  --enso-docs-code-background-color: #dddcde;
+  font-family: 'M PLUS 1', DejaVuSansMonoBook, sans-serif;
+  font-size: 11.5px;
+  line-height: 160%;
+  color: var(--enso-docs-text-color);
+  background-color: var(--enso-docs-background-color);
+  padding: 8px 8px 4px;
+  white-space: normal;
+}
+
+:deep(.sectionContent) {
+  padding: 0 8px;
+}
```

Comment:
I'm not a fan of this. The styles are scoped for a reason. Is the `deep` really necessary here?

Also, this probably should be a margin. Margins are usually logically considered to be "managed by the parent elements". CSS kinda took that more right with the grid `gap` property. Here, I would either use `gap`s, or at least move the class declaration to the component that styles it. Note that vue [allows declaring classes or other properties on components](https://vuejs.org/guide/essentials/class-and-style.html#with-components) in a template.

Interesting article for more context: [**Axiomatic CSS and Lobotomized Owls**](https://alistapart.com/article/axiomatic-css-and-lobotomized-owls/)
I've used that technique previously, and it works pretty well for global styles, but isn't that great with vue scopes (which are just [fancy CSS selector macros](https://vuejs.org/api/sfc-css-features.html#scoped-css)). Now we should probably use `gap`s for similar behavior. But the idea of "who really owns that property" still stands.

Reply:
I removed this particular instance of `:deep`, it was not needed after a refactoring. Thanks for an interesting read, I don‘t think it should be applied here. I refactored the layout a bit and managed to achieve the needed result much easier with a single additional div in DocumentationPanel.